### PR TITLE
Proper client close

### PR DESCRIPTION
**Pull request by andrewvc**
This properly closes connections on shutdown as well as prepping this plugin for the ng_pipeline by defining a `multi_receive` method.

### CODE REVIEW

**Comment by jsvd** [comment]
@andrewvc would you consider just creating a PR that adds the close method and another for the ng pipeline?

**Comment by suyograo** [comment]
+1. that way we can get the `#close` fix to 2.1.2 and then prep for ng-pipeline separately.

**Comment by andrewvc** [comment]
@jsvd @suyograo I didn't realize this was a bug in 2.1.x, I thought we were targeting 2.2. That being said, while that's my mistake, it would be a decent amount of extra effort to untangle this stuff into two PRs. The code as is supports 2.1.x just fine. Would you guys mind reviewing the combined code just this once? I mean, the close patch is just `def close; client.close; end`. It has a dependency on the other stuff however in that it must switch to using `#background` vs. the previous async stuff.

**Comment by jsvd** [comment]
@andrewvc it's not  bug indeed, it will only be useful when restarting a pipeline, I was just advocating for the general principle of small PRs.
That said I'll review as is :+1:

**Comment by andrewvc** [comment]
@jsvd ah, yes, that's why I did the refactor inline. I felt it qualified for a 2.2.x target because there was no way to upgrade to manticore 0.5.2 without refactoring the function, and if we were targetting 2.2.x the correct implementation would be a `multi_receive` default with a `receive` fallback. That being said, this is a bit of a gray area in the one fix per PR thing. I could see myself arguing either side.

File: lib/logstash/outputs/http.rb [hunk 1]
```diff
@@ -89,31 +89,29 @@ def register
...
-  # TODO: (colin) the request call sequence + async handling will have to be reworked when using
-  # Maticore >= 5.0. I will set a version constrain in the gemspec for this.
-  def receive(event)
+  def multi_receive(events)
+    events.each {|event| receive(event, :parallel)}
+    client.execute!
+  end
+
+  # Once we no longer need to support Logstash < 2.2 (pre-ng-pipeline)
+  # We don't need to handle :background style requests
```
Comment (jsvd) [inline lib/logstash/outputs/http.rb]: I would appreciate a more detailed information on what request_proxy_type background/parallel actually does.
the name `request_proxy_type` makes me think of types of "request proxies"? *feels lost and confused*

File: lib/logstash/outputs/http.rb [hunk 2]
```diff
@@ -138,8 +136,11 @@ def receive(event)
       )
     end
 
-    # Invoke it using the Manticore Executor (CachedThreadPool) directly
-    request_async_background(request)
```
Comment (jsvd) [inline lib/logstash/outputs/http.rb]: if this is no longer used the method can be removed as well

File: lib/logstash/outputs/http.rb (hunk 1 above)
Comment (andrewvc) [inline lib/logstash/outputs/http.rb]: Well, that's a manticore-ism. In manticore client.background returns a backgrounding proxy. It's not an HTTP proxy, but a proxy object. In this context I should rename it to `request_async_type` for clarity.

File: lib/logstash/outputs/http.rb (hunk 2 above)
Comment (andrewvc) [inline lib/logstash/outputs/http.rb]: +1

File: lib/logstash/outputs/http.rb (hunk 1 above)
Comment (jsvd) [inline lib/logstash/outputs/http.rb]: :+1: on the rename

**Comment by andrewvc** [comment]
@jsvd I've clarified the code (and added some clarifying comments as well). How's it look now?

I'm out on vacation as of tomorrow. If it passes your review would you mind merging and publishing the new gem?

**Pull request merged**